framework/BaseClasses/BaseInterface.py

```diff
@@ -0,0 +1,222 @@
+
+  def readXML(self, xmlNode, variableGroups={}, globalAttributes=None):
+    """
+      provide a basic reading capability from the xml input file for what is common to all types in the simulation than calls _readMoreXML
+      that needs to be overloaded and used as API. Each type supported by the simulation should have: name (xml attribute), type (xml tag),
+      verbosity (xml attribute)
+      @ In, xmlNode, ET.Element, input xml
+      @ In, variableGroups, dict{str:VariableGroup}, optional, variable groups container
+      @ In, globalAttributes, dict{str:object}, optional, global attributes
+      @ Out, None
+    """
+    self.variableGroups = variableGroups
+    if 'name' in xmlNode.attrib.keys():
+      self.name = xmlNode.attrib['name']
+    else:
+      self.raiseAnError(IOError,'not found name for a '+self.__class__.__name__)
+    self.type = xmlNode.tag
+    if globalAttributes is not None:
+      self.globalAttributes = globalAttributes
+    if 'verbosity' in xmlNode.attrib.keys() or 'verbosity' in self.globalAttributes:
+      verbGlobal = None if self.globalAttributes is None else self.globalAttributes.get('verbosity')
+      verbLocal = xmlNode.attrib.get('verbosity')
+      self.verbosity = verbLocal if verbLocal is not None else verbGlobal
+      self.raiseADebug('Set verbosity for '+str(self)+' to '+str(self.verbosity))
+    self._readMoreXML(xmlNode)
+    self.raiseADebug('------Reading Completed for:')
+    self.printMe()
+
+  def handleInput(self, paramInput, variableGroups={}, globalAttributes=None):
+    """
+      provide a basic reading capability from the xml input file for what is common to all types in the simulation than calls _handleInput
+      that needs to be overloaded and used as API. Each type supported by the simulation should have: name (xml attribute), type (xml tag),
+      verbosity (xml attribute)
+      @ In, paramInput, InputParameter, input data from xml
+      @ In, variableGroups, dict{str:VariableGroup}, optional, variable groups container
+      @ In, globalAttributes, dict{str:object}, optional, global attributes
+      @ Out, None
+    """
+    self.variableGroups = variableGroups
+    if 'name' in paramInput.parameterValues:
+      self.name = paramInput.parameterValues['name']
+    else:
+      self.raiseAnError(IOError,'not found name for a '+self.__class__.__name__)
+    self.type = paramInput.getName()
+    if self.globalAttributes is not None:
+      self.globalAttributes = globalAttributes
+    if 'verbosity' in paramInput.parameterValues:
+      self.verbosity = paramInput.parameterValues['verbosity'].lower()
+      self.raiseADebug('Set verbosity for '+str(self)+' to '+str(self.verbosity))
+    self._handleInput(paramInput)
+    self.raiseADebug('------Reading Completed for:')
+    self.printMe()
+
+  def _readMoreXML(self,xmlNode):
```

Comment:
Hopefully, yes. It will take converting existing Entities to using the handleInput pattern, then this can be removed and each Entity or Interface can extend handleInput.